pkg/token/token.go

```diff
@@ -392,6 +392,18 @@ type tokenVerifier struct {
 	validSTShostnames map[string]bool
 }
 
+func getDefaultHostNameForRegion(partition *endpoints.Partition, region string) (string, error) {
+	rep, err := partition.EndpointFor(stsServiceID, region, endpoints.STSRegionalEndpointOption, endpoints.ResolveUnknownServiceOption)
```

Comment:
qs for familiarity:  Looks like `endpoints.ResolveUnknownServiceOption` will return an error [here[(https://github.com/aws/aws-sdk-go/blob/fbe26bf118cf10a5c6aef4a105e2a5609c617b1a/aws/endpoints/v3model.go#L218-L221). What was the current behavior of `partition.EndpointFor` when `endpoints.ResolveUnknownServiceOption` was false?

Reply:
currently it would throw service not found https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/pkg/token/token.go#L412-L413 as the services are not yet updated in the sdk go

Reply:
discussed offline, will add an unit test here